extensions/ql-vscode/src/vscode-tests/no-workspace/history-item-label-provider.test.ts

```diff
@@ -126,7 +126,35 @@ describe('HistoryItemLabelProvider', () => {
       expect(labelProvider.getShortLabel(fqi)).to.eq('query-name');
     });
 
-    function createMockRemoteQueryInfo(userSpecifiedLabel?: string, repositoryCount?: number) {
+    describe('when results are present', () => {
+      it('should display results if there are any', () => {
+        const fqi = createMockRemoteQueryInfo({ resultCount: 16, repositoryCount: 2 });
+
+        config.format = '%t %q %d %s %f %r %%';
+        expect(labelProvider.getLabel(fqi)).to.eq(`${dateStr} query-name (javascript) 2 repositories in progress query-file.ql (16 results) %`);
+      });
+    });
+
+    describe('when results are not present', () => {
+      it('should skip displaying them', () => {
+        const fqi = createMockRemoteQueryInfo({ resultCount: 0, repositoryCount: 2 });
+
+        config.format = '%t %q %d %s %f %r %%';
+        expect(labelProvider.getLabel(fqi)).to.eq(`${dateStr} query-name (javascript) 2 repositories in progress query-file.ql  %`);
+      });
+    });
+
+    interface mockRemoteQueryInterface {
```

Comment:
Interface names should start with a capital letter.

I'm not sure why we're defining an interface here though, we could just say that the return type of `createMockRemoteQueryInfo` is `RemoteQueryHistoryItem`.

BTW (and this is nothing to do with your changes, I'm just sharing some thoughts) I'm not very keen on the approach we've been taking around "forcing" a type on an object when it's not entirely true. I think it can lead to confusing code. What I mean with that is that we do `as unknown as RemoteQueryHistoryItem` (and not just `as RemoteQueryHistoryItem`), presumably because the object doesn't define all the fields it needs in order to be a proper `RemoteQueryHistoryItem`. Then the user of this object will assume it's a proper `RemoteQueryHistoryItem` and could fail if trying to access one of its values. I think that's confusing from an engineer's perspective. We should set all the fields instead. If that's a pain then we need to build helpers around creating these objects.

Reply:
> Interface names should start with a capital letter.

Thanks 👍 

> we could just say that the return type of createMockRemoteQueryInfo is RemoteQueryHistoryItem.

In this case we were setting `userSpecifiedLabel` and `repositoryCount` which aren't present on `RemoteQueryHistoryItem`. 

But I agree this could be a lot cleaner.

Reply:
(feel free to merge the PR, just some thoughts from me - we could tidy up in a future PR)

 I'm really not following what this object is supposed to be. It seems to have a bunch of properties - some from `RemoteQueryHistoryItem` and some not. The function under test is `getLabel` and that takes a `QueryHistoryInfo` so I think that's what this object is supposed to be, but it has a few issues. I think the following would remove type errors, but I'm unsure why `in progress` is expected and not `InProgress` so that will likely fail:

```
function createMockRemoteQueryInfo({
  resultCount = 16,
  userSpecifiedLabel = undefined,
  repositoryCount = 0
}: {
  resultCount?: number;
  userSpecifiedLabel?: string;
  repositoryCount?: number;
}): QueryHistoryInfo {
  return {
    t: 'remote',
    userSpecifiedLabel,
    remoteQuery: {
      executionStartTime: date.getTime(),
      queryName: 'query-name',
      queryFilePath: 'query-file.ql',
      controllerRepository: {
        owner: 'github',
        name: 'vscode-codeql-integration-tests'
      },
      language: 'javascript',
      actionsWorkflowRunId: 123,
      queryText: 'query-text',
      repositoryCount,
    },
    status: QueryStatus.InProgress,
    resultCount,
    completed: false,
    queryId: 'query-id'
  };
}
```

Reply:
Thank for this @charisk! 

I've added `QueryHistoryInfo` as a return type without the extra fields and the tests run fine. 

Oddly it didn't complain about missing fields. 🤔